pkg/operator/ceph/cluster/controller.go

```diff
@@ -118,6 +120,8 @@ func newReconciler(mgr manager.Manager, ctx *clusterd.Context, clusterController
 		panic(err)
 	}
 
+	clusterController.recorder = eventutil.NewEventReporter(mgr.GetEventRecorderFor("ClusterController"), 10, 5)
```

Comment:
What is the goal of reporting the events? Seems like it is to report both failures and successes of a reconcile. I wonder if we can report the events at the same time as we are updating the conditions. The conditions are where we have more detailed status of the reconcile. Perhaps we could report an event for all the condition updates except the "Progress" updates that are during a reconcile. There are a lot of changes to conditions coming with #7315, so please wait for that to merge before you look into any related changes.

Reply:
@iamniting what's the status of this PR? Did you get a chance to revisit it with the above comment?

Reply:
Most of it is done, Just waiting for @travisn replies. So we can take a decision to implement it either at one place or many places like conditions.